src/GitTools.jl

```diff
@@ -274,7 +274,7 @@ function tree_hash(::Type{HashType}, root::AbstractString; debug_out::Union{IO,N
     entries = Tuple{String, Vector{UInt8}, GitMode}[]
     for f in sort(readdir(root; join=true); by = f -> isdir(f) ? f*"/" : f)
         # Skip `.git` directories
-        if f == ".git"
+        if basename(f) == ".git"
```

Comment:
I thought `Tar.jl` could have been the culprit since the issue was reported only on nightly but on v1.6, and `Tar.jl` had recent changes in permissions.  Interestingly enough, this line was introduced [two years ago](https://github.com/JuliaLang/Pkg.jl/commit/863132441d4be524053f5580012837389a569bb7), so how the error isn't reproducible on v1.6? :confused:

Reply:
The original version didn't have this bug. The problem is that the new version passes `join=true` to `readdir` which means that an entry named `.git` doesn't match this check.